lib/customer.rb

```diff
@@ -0,0 +1,74 @@
+require_relative 'order'
+require_relative 'menu'
+require_relative 'twilio'
+
+class Customer
+  attr_accessor :name, :phone_number, :order, :selection, :price, :total
+  def initialize(order = Order.new)
+    @name
+    @phone_number
+    @order = order
+    @selection
+    @quantity
+    @price
+    @total
+  end
+
+  def place_order
+    get_name
+    @order.select_dishes(@selection, @price, @quantity)
+    get_total
+  end
+
+  def get_name
+    puts "Good evening, can I get your name please?"
+    @name = gets.chomp
+    get_phone
+  end
+
+  def get_phone
+    puts "and what is your phone number, for delivery verification?"
+    @phone_number = gets.chomp.to_i
+    see_menu
+  end
+
+  def see_menu
+    self.order.menu.display.each { |k,v| puts "#{k}:#{v}" }
+    get_food
+  end
+
+  def get_food
+    puts "What will you be having today?"
+    @selection = gets.chomp.downcase
+    raise "That dish is not on the menu!" if @order.menu.display.has_key?(@selection) == false
+    get_quantity
```

Comment:
I think these methods are very interdependent (they all call one another in succession). That's not really OOP, it's really functional chaining. You don't want to do that in OOP, or in Ruby to be honest.

Make sure all these methods can be called separately, then use a method to call them all one after the other if needed.